### ISSUE TITLE
Update Rule(s): Multiple rules need this assumption

### ISSUE THREAD

**Issue by jkodu** [auto-wcag/auto-wcag#362]
Rules to be updated:
- https://auto-wcag.github.io/auto-wcag/rules/SC4-1-2-role-attribute-has-valid-value.html
- https://auto-wcag.github.io/auto-wcag/rules/SC4-1-2-aria-state-or-property-has-valid-value.html
- https://auto-wcag.github.io/auto-wcag//rules/SC4-1-1+SC4-1-2-aria-state-or-property-allowed.html

There are various scenario's where an invalid value for role is used on an element. 

Believe this is not a WCAG failure. Ideally we should update the rule to have an assumption saying - `The role is being used to comply to WCAG`.

Some scenarios:

1. In this case `aria-expanded` is used to represent a state and hence used to comply to WCAG.
`<div role='radio' aria-checked=false aria-expanded=false></div>`

2. In this case `role` is not used to comply to WCAG.
`<header role='header'>`

**Comment by annethyme** [comment]
Related to this: https://github.com/auto-wcag/auto-wcag/issues/353

**Comment by jkodu** [comment]
PR in place, closing.
